### PR TITLE
Add https support

### PR DESCRIPTION
**Pull request by jonasfj**
Ideally we shouldn't add HTTPS support here... But it's so easy to do.

In fact even when we setup a reverse HTTPS proxy for other things that would like to have  this, tasks offering things, etc.

We might want to keep HTTPS implemented this way. That way it's not coupled with all the other network routing we have to do... Configuring an nginx proxy is hard enough without adding more special cases...

### CODE REVIEW

**Comment by jonasfj** [comment]
Okay, also added the token stuff... It was easy...

I would like to see this multiplexing too... But that would make the access token harder to implement.
Granted it would mean less memory overhead on the servers...

**Comment by lightsofapollo** [comment]
@jonasfj lgtm I ran the tests locally ... I think I missed updating the paths when I moved the project so before landing please apply this https://gist.github.com/lightsofapollo/abf586640753fbdbb9d0

File: main.go [hunk 1]
```diff
@@ -28,11 +29,23 @@ func abort(writer http.ResponseWriter) error {
...
 	routes := http.NewServeMux()
-	routes.HandleFunc("/log", func(w http.ResponseWriter, r *http.Request) {
+	routes.HandleFunc("/log/", func(w http.ResponseWriter, r *http.Request) {
 		debug("output %s %s", r.Method, r.URL.String())
-		// TODO: Add method context switching here...
-		getLog(stream, w, r)
+
+		// Authenticate the request with accessToken, this is good enough because
+		// live logs are short-lived
+		if (r.URL.String()[5:] != accessToken) {
```
Comment (petemoore) [inline main.go]: maybe just a comment above this line to explain the r.URL.String() is "/log/" + accessToken, so skip first 5 chars?

**Comment by petemoore** [comment]
looks good to me too! can't really comment on the node tests, but the go code looks good. thanks!

**Pull request merged**